quil-rs/src/program/memory.rs

```diff
@@ -342,9 +342,10 @@ impl Expression {
     pub fn get_memory_references(&self) -> Vec<&MemoryReference> {
         match self {
             Expression::Address(reference) => vec![reference],
-            Expression::FunctionCall(FunctionCallExpression { expression, .. }) => {
-                expression.get_memory_references()
-            }
+            Expression::FunctionCall(FunctionCallExpression { arguments, .. }) => arguments
```

Comment:
I think somewhere we should explicitly reference the quote you pulled from the spec:

> Extern CALLs may appear in arithmetic expressions with some restrictions: the extern MUST have a declared function signature, which MUST include a return type, and which MUST NOT include any mutable parameters.

This is why we can simply return `Vec<MemoryReference>` here (all presumed to be reads), while `Call::get_memory_accesses` returns a `MemoryAccessesResult`.